pkg/sdk/go/users_test.go

```diff
@@ -181,3 +400,129 @@ func TestCreateToken(t *testing.T) {
 		assert.Equal(t, tc.token, token, fmt.Sprintf("%s: expected response: %s, got:  %s", tc.desc, token, tc.token))
 	}
 }
+
+func TestUpdateUser(t *testing.T) {
+	svc := newUserService()
+	ts := newUserServer(svc)
+	defer ts.Close()
+	sdkConf := sdk.Config{
+		UsersURL:        ts.URL,
+		MsgContentType:  contentType,
+		TLSVerification: false,
+	}
+
+	mainfluxSDK := sdk.NewSDK(sdkConf)
+	user := sdk.User{Email: "user@example.com", Password: "password"}
+
+	mockAuthzDB := map[string][]mocks.SubjectSet{}
+	mockAuthzDB[user.Email] = append(mockAuthzDB[user.Email], mocks.SubjectSet{Object: "authorities", Relation: "member"})
+	auth := mocks.NewAuthService(map[string]string{user.Email: user.Email}, mockAuthzDB)
+
+	tkn, _ := auth.Issue(context.Background(), &mainflux.IssueReq{Id: user.ID, Email: user.Email, Type: mfauth.APIKey})
+	token := tkn.GetValue()
+	userID, err := mainfluxSDK.CreateUser(token, user)
+	require.Nil(t, err, fmt.Sprintf("unexpected error: %s", err))
+	usertoken, err := mainfluxSDK.CreateToken(user)
+	require.Nil(t, err, fmt.Sprintf("unexpected error: %s", err))
+
+	cases := []struct {
+		desc  string
+		user  sdk.User
+		token string
+		err   error
+	}{
+		{
+			desc:  "update email for user",
+			user:  sdk.User{ID: userID, Email: "user2@example.com", Password: "password"},
+			token: usertoken,
+			err:   nil,
+		},
+		{
+			desc:  "update email for non existing user",
+			user:  sdk.User{ID: "0", Email: "user2@example.com", Password: "password"},
+			token: wrongValue,
```

Comment:
This case should have a valid token so that it passes authorization.

Reply:
This is returning nil. I don't think it is correct. Should it return, `ErrFailedFetch`?

Reply:
With `cli` it is able to update while with the test it fails. I will look into the mock and check what is the problem

Reply:
If the token is valid, it will return the useridentity which has a valid email address. The valid email address will pass our mock tests. If we pass an invalid token it will fail to pass as you are `Unauthorized`. This means when we are updating the user it doesn't matter the value of userID or userEmail but rather the token to get the actual user as it uses the Email and ID from the token.

Reply:
This effectively means that the user can update only its own metadata. If that's the case, please update all the tests (and service if needed) to match the actual endpoint and what it does.